### PR TITLE
go/store/prolly/{message, tree}: Reduce memory footprint of `message.ItemAccess`, `tree.Node`

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
- Refactored `message.ItemAccess` (previously `ItemArray`) to decrease memory footprint from 48 bytes to 10.
- Refactored `tree.Node` to decrease memory footprint from 152 bytes to 56
- `tree.nodeCache` now caches `tree.Node` instead of `chunks.Chunk`. This eliminates decode time for cache hits.
- Reworked lock striping to `tree.nodeCache` to improved access latency.

### CODE REVIEW

**Comment by andrew-wm-arthur** [comment]
#benchmark

File: go/store/prolly/message/merge_artifacts.go [hunk 1]
```diff
@@ -62,7 +66,7 @@ func (s MergeArtifactSerializer) Serialize(keys, values [][]byte, subtrees []uin
 	keySz, valSz, bufSz := estimateMergeArtifactSize(keys, values, subtrees, s.keyDesc.AddressFieldCount())
 	b := getFlatbufferBuilder(s.pool, bufSz)
 
-	// serialize keys and offsets
+	// serialize keys and offStart
```
Comment (max-hoffman) [inline go/store/prolly/message/merge_artifacts.go:69]: maybe accidental refactor, also below

File: go/store/prolly/tree/node_cache.go [hunk 2]
```diff
@@ -15,45 +15,42 @@
...
 	"sync"
 
-	"github.com/dolthub/dolt/go/store/chunks"
 	"github.com/dolthub/dolt/go/store/hash"
 )
 
 const (
-	numStripes = 256
+	numStripes      = 32
+	stripeMask byte = 0b00011111
```
Comment (max-hoffman) [inline go/store/prolly/tree/node_cache.go:26]: is this an alternative to hashing or mod? a comment might be helpful

**Review by max-hoffman — APPROVED** [review]
LGTM! definitely a grab bag but it all the improvements look nice

**Comment by max-hoffman** [comment]
Also @timsehn this brings makes new format groupby faster than old, if you hadn't already seen

**Comment by timsehn** [comment]
OOOO. I had not seen! Beautiful

**Pull request merged**